src/observables/ForkJoinObservable.ts

```diff
@@ -3,41 +3,41 @@ import {Subscriber} from '../Subscriber';
 import {PromiseObservable} from './PromiseObservable';
 import {EmptyObservable} from './EmptyObservable';
 import {isPromise} from '../util/isPromise';
+import {isArray} from '../util/isArray';
 
 export class ForkJoinObservable<T> extends Observable<T> {
-  constructor(private sources: Array<Observable<any> |
-                                    Promise<any> |
-                                    ((...values: Array<any>) => any)>) {
-     super();
+  constructor(private sources: Array<Observable<any> | Promise<any>>,
+              private resultSelector?: (...values: Array<any>) => any) {
+    super();
   }
 
   static create(...sources: Array<Observable<any> |
+                                  Array<Observable<any>> |
                                   Promise<any> |
-                                  ((...values: Array<any>) => any)>)
-                                  : Observable<any> {
-    if (sources === null || sources.length === 0) {
+                                  ((...values: Array<any>) => any)>) : Observable<any> {
+    if (sources === null || arguments.length === 0) {
       return new EmptyObservable();
     }
-    return new ForkJoinObservable(sources);
-  }
 
-  private getResultSelector(): (...values: Array<any>) => any {
-    const sources = this.sources;
+    let resultSelector: (...values: Array<any>) => any = null;
+    if (typeof sources[sources.length - 1] === 'function') {
+      resultSelector = <(...values: Array<any>) => any>sources.pop();
+    }
 
-    let resultSelector = sources[sources.length - 1];
-    if (typeof resultSelector !== 'function') {
-      return null;
+    // if the first and only other argument besides the resultSelector is an array
+    // assume it's been called with `forkJoin([obs1, obs2, obs3], resultSelector)`
+    if (sources.length === 1 && isArray(sources[0])) {
+      sources = <Array<Observable<any>>>sources[0];
     }
```

Comment:
Seems previous comment went away by updating. Appreciate @blesh for suggestion, it seems much better than I did first time. :)